### ISSUE TITLE
roachtest: test multi-statement transactions do not need retries

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#36152]
From a conversation with @nvanbenschoten , we realized that we need to confirm that a multi-statement transaction, issued all at once in a single semi-colon separated string, never needs client-side retries. 

It shouldn't, but @nvanbenschoten isn't sure if we test specifically for it and we think we should verify this via internal testing.

**Comment by nvanbenschoten** [comment]
@andreimatei this is still true even with the new `connExecutor` structure if we don't overfill the response buffer and force results to be streamed, right?

**Comment by petermattis** [comment]
This would presumably be a unit test, not a roachtest, right?

**Comment by nvanbenschoten** [comment]
> This would presumably be a unit test, not a roachtest, right?

Yes, I reworded this a little and kicked it over to the SQL team.

**Comment by rytaft** [comment]
@rafiss any reason you think this needs to be on the SQL Queries board too? I'm going to remove our project but feel free to add it back if you think we can help with this.

**Comment by rafiss** [comment]
@rytaft My reasoning was that the former SQL Execution team was the main group working on the `connExecutor` logic where this test would need to be done, so I thought SQL Queries might be a good place for this. But since SQL Experience will be working on transaction retry guidance this release, I'm fine for us to take this on initially and reach out if needed.

**Comment by andreimatei** [comment]
> @andreimatei this is still true even with the new connExecutor structure if we don't overfill the response buffer and force results to be streamed, right?

Right.

**Comment by andreimatei** [comment]
I think the test this issue is asking for is here:
https://github.com/cockroachdb/cockroach/blob/5b7a6102a134c657ebb7836d6f134842ea78d8dd/pkg/sql/txn_restart_test.go#L521
